lib/widgets/whole_text_field.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:flutter/material.dart';
+
+class WholeTextField extends StatefulWidget {
```

Comment:
i tested it and the `TextField` class automatically expands (width) according to its parent

![image](https://user-images.githubusercontent.com/72055285/116956145-170b0680-acc7-11eb-9eeb-137dab149ab2.png)

to reduce code, you can simply merge all three classes into one:

```
class ??? extends StatefulWidget
  accept a boolean if it's multilined or not
  accept an integer for maxlength
}

class ??? state
  return Container() -- for box decoration with child of TextInput
  -- never add padding for singular widgets, let the page code handle that
```

Reply:
Alright I'll try this

Reply:
alright, that should be the last commit or test for this branch. i'll fix the conflicts asap (after your commit) so that we can merge it to develop!

please don't create a new branch before i merge it

Reply:
Done! Thanks for the advice :>

![image](https://user-images.githubusercontent.com/72382027/116964668-4ecd7a80-acd6-11eb-92c7-320b3edd1c9c.png)

Reply:
When this is merged I'll pull from develop and work on the other widgets